scripts/fabtests/ccm_runfabtests.sh

```diff
@@ -0,0 +1,196 @@
+ccm_login="ccmlogin"
+ccm_login_script="${HOME}/ccm_login_fabtests_script.$$"
+fabtests_directory="`pwd`"
+fabtests_cmd="./runfabtests.sh -p "
+provider="gni"
+test_suite=""
+use_parameters=0
+verbose=""
+
+usage() {
+    ec=$1
+    echo "cray_ccm_runfabtests.sh [-d <directory>] [-D] [-p <provider>] [-t <test_suite>] [-u] [-v] [-vv] [-vvv]"
+    echo "   -d libfabric tests location"
+    echo "   -D print debug output"
+    echo "   -p libfabric provider to use"
+    echo "   -t libfabric test suite to execute"
+    echo "   -u use parameters on runfabtests"
+    echo "   -v verbose mode for runfabtests"
+    exit $ec
+}
+
+if [ $# -gt 0 ] ; then
+  while getopts "d:Dhp:t:uv:" option; do
+    case $option in
+      d) fabtests_directory=$OPTARG;;
+      D) debug=1;;
+      h) usage 0 ;;
+      p) provider="$OPTARG";;
+      t) test_suite="-t $OPTARG";;
+      u) use_parameters=1;;
+      v) verbose="${verbose} -v";;
+      *) usage 1;;
+    esac
+  done
+fi
+
+#
+# Check for ccmlogin
+#
+ccmlogin=`command -v ccmlogin`
+if [ $? != 0 ]; then
+    ${load_ccm_module}
+    if [ $? != 0 ]; then
+        echo "Error: CCM Module not loaded.  Can Not execute the FAB tests."
+        exit 1
+    fi
+
+    ccmlogin=`command -v ccmlogin`
+    if [ $? != 0 ]; then
+        echo "Error: CCM Login command not found.  Can Not execute the FAB tests."
+        exit 1
+    fi
+fi
+
+rm -f ${ccm_login_script}
+cat > ${ccm_login_script} <<"END_OF_CCMLOGIN_SCRIPT_1"
+#!/bin/bash
+
+END_OF_CCMLOGIN_SCRIPT_1
+
+echo "debug=${debug}" >> ${ccm_login_script}
+echo "fabtests_directory='${fabtests_directory}'" >> ${ccm_login_script}
+echo "provider='${provider}'" >> ${ccm_login_script}
+echo "test_suite='${test_suite}'" >> ${ccm_login_script}
+echo "use_parameters='${use_parameters}'" >> ${ccm_login_script}
+echo "verbose='${verbose}'" >> ${ccm_login_script}
+
+cat >> ${ccm_login_script} <<"END_OF_CCMLOGIN_SCRIPT_2"
+node_1=""
+node_2=""
+
+if [[ "${provider}" == "gni" ]]
+then
+    ccm_login_node_dir="${HOME}/.crayccm/"
+    ccm_nodelist_file="${ccm_login_node_dir}/`ls ${ccm_login_node_dir}`"
+    node_index=0
+
+    if [[ ${debug} -ne 0 ]]
+    then
+        echo "*** pid: $$, ccm_nodelist_file: '${ccm_nodelist_file}'"
+    fi
+
+    while read -r node_entry
+    do
+        if [[ ${debug} -ne 0 ]]
+        then
+            echo "*** pid: $$, next node_index: '${node_index}'";
+            echo "*** pid: $$, next node_entry: '${node_entry}'";
+        fi
+        if [[ ${node_index} -eq 0 ]]
+        then
+            if [[ ${use_parameters} -eq 0 ]]
+            then
+                node_1="${node_entry}"
+            else
+                node_1="-s ${node_entry}"
+            fi
+            node_index=1
+            if [[ ${debug} -ne 0 ]]
+            then
+                echo "*** pid: $$, set node_1: '${node_1}'";
+            fi
+        else
+            if [[ "${node_2}" == "" ]]
+            then
+                if [[ "${node_1}" != "${node_entry}" ]]
```

Comment:
Yes, this is used to find 2 unique nids.
I have added code to use node_1 as the second node, if a second node was not found.